### ISSUE TITLE
OAuth and redirection

### ISSUE THREAD

**Issue by barryo** [peeringdb/peeringdb#530]
I'm (finally) implementing OAuth against PeeringDB for IXP Manager.

If I have an active session with PeeringDB and I try to log into IXP Manager via PeeringDB OAuth it works fine.

If I try to log into IXP Manager via PeerginDB OAuth **but I have to log into PeeringDB as part of the process flow**, I do not get either redirected back to IXP Manager or the *Application requires following permissions* after login but rather get the normal post-login PeeringDB landing page.

In other words, PeeringDB does not seem to store and use the requested action if a login is required.

**Comment by barryo** [comment]
Hey @grizz - I meant to add, if the above does not make sense, I can demo via Skype screenshare or something.

**Comment by grizz** [comment]
@barryo awesome!

I believe this a dupe of #482?

**Comment by barryo** [comment]
Opps, sure is. Sorry, never saw that. I guess no demo required then. I'll close this.

I'm essentially all good to release a version of IXP Manager with this but I can't really until #482 is fixed as the flow is too confusing. I.e.

1. Goto IXP Manager
2. Click login with PeeringDB
3. Log into PeeringDB
4. Get post-login PeeringDB welcome page
5. Reset browser URL to IXP Manager
6. repeat (1) and (2) again to now get authorize IXP Manager (or redirected to IXP manager logged in if already auth'd).

**Comment by job** [comment]
I think we should consider this a high priority

**Comment by grizz** [comment]
@job I agree, go +1 #482 :)

**Comment by thatchrisp** [comment]
Also +1 #482.
